pkg/providers/tinkerbell/hardware/machine.go

```diff
@@ -0,0 +1,121 @@
+package hardware
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+)
+
+// Machine is a machine configuration with optional BMC interface configuration.
+type Machine struct {
+	// Id uniquely identifies this machine. It is programmatically generated so is not readable from CSV.
+	Id          string      `csv:"-"`
```

Comment:
So during e2e tests the Id already exists in Tink stack so we do a dry run and pass the id that matches the id in tink. So in that case guid header will be in the csv. Will this account for that?

Reply:
I ask so I can use the same struct in e2e vs creating a different one. If I need to keep it different that is fine too I guess. Whatever you recommend.

Reply:
This is where tags become a little annoying. In some contexts you want it to be required and others you don't but tags aren't dynamic like that.

Do you need the ID in the CSV or could you generate it?

Reply:
We decided the tests can use a wrapping construct to read the CSV and require a UUID.